warehouse/utils/security_policy.py

```diff
@@ -99,8 +100,10 @@ def permits(self, request, context, permission):
             if isinstance(identity, User):
                 principals.append(f"user:{identity.id}")
                 principals.extend(_principals_for_authenticated_user(identity))
+            elif isinstance(identity, OIDCProvider):
+                principals.append(f"oidc:{identity.id}")
```

Comment:
Same note as https://github.com/pypi/warehouse/pull/11272/files#r923830262 -- this is the right principal to apply, but it's probably effectively a no-op since we don't match against ACLs for Macaroons at the moment.